app/src/main/java/com/pvp/app/ui/screen/goals/GoalScreen.kt

```diff
@@ -0,0 +1,338 @@
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.LinearProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.pvp.app.model.Goal
+import com.pvp.app.ui.screen.layout.FloatingActionButton
+
+@Composable
+fun GoalScreen(
+    model: GoalViewModel = hiltViewModel(),
+    modifier: Modifier
+) {
+    val state by model.state.collectAsStateWithLifecycle()
+    var isDialogOpen by remember { mutableStateOf(false) }
+    val toggleDialog = remember { { isDialogOpen = !isDialogOpen } }
+
+    Scaffold(
+        content = { paddingValues ->
+            Box(
+                modifier = modifier
+                    .fillMaxSize()
+                    .padding(paddingValues)
+                    .padding(16.dp)
+            ) {
+                Column {
+                    val goals = state.currentGoals
+
+                    val filter by remember {
+                        derivedStateOf {
+                            if (state.monthly) GoalFilter.Monthly else GoalFilter.Weekly
+                        }
+                    }
+
+                    GoalTypeFilter(filter = filter) {
+                        if (it != filter) model.changeMonthly()
+                    }
+
+                    Spacer(modifier = Modifier.padding(8.dp))
+
+                    DateChanger()
+
+                    Spacer(modifier = Modifier.padding(8.dp))
+
+                    LazyColumn(modifier = Modifier.fillMaxWidth()) {
+                        if (!goals.any()) {
+                            item {
+                                Box(
+                                    contentAlignment = Alignment.Center,
+                                    modifier = Modifier.fillMaxWidth()
+                                ) {
+                                    Text(
+                                        fontStyle = FontStyle.Italic,
+                                        modifier = Modifier.padding(32.dp),
+                                        style = MaterialTheme.typography.bodyMedium,
+                                        text = "No goals have been set up yet."
```

Comment:
We may want to dispose the `.` at the end, since everywhere else (friend requests; tasks list; decorations) we are providing this text as title/label and not in sentence form (which would require punctuation).